components/org.wso2.carbon.identity.application.authenticator.requestpath.basicauth/src/test/java/org/wso2/carbon/identity/application/authenticator/requestpath/basicauth/BasicAuthRequestPathAuthenticatorTest.java

```diff
@@ -0,0 +1,242 @@
+
+import org.apache.axiom.om.util.Base64;
+import org.apache.commons.logging.Log;
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.stubbing.Answer;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.testng.IObjectFactory;
+import org.mockito.Mock;
+import org.testng.annotations.*;
+import org.wso2.carbon.identity.application.authentication.framework.context.AuthenticationContext;
+
+import static org.mockito.Matchers.any;
+import static org.mockito.MockitoAnnotations.initMocks;
+import static org.powermock.api.mockito.PowerMockito.*;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertNull;
+
+import org.wso2.carbon.identity.application.authentication.framework.exception.AuthenticationFailedException;
+import org.wso2.carbon.identity.application.authentication.framework.exception.InvalidCredentialsException;
+import org.wso2.carbon.identity.application.authentication.framework.model.AuthenticatedUser;
+import org.wso2.carbon.identity.application.authentication.framework.util.FrameworkUtils;
+import org.wso2.carbon.identity.application.authenticator.requestpath.basicauth.internal.BasicAuthRequestPathAuthenticatorServiceComponent;
+import org.wso2.carbon.identity.application.common.model.User;
+import org.wso2.carbon.identity.core.util.IdentityTenantUtil;
+import org.wso2.carbon.identity.core.util.IdentityUtil;
+import org.wso2.carbon.identity.testutil.powermock.PowerMockIdentityBaseTest;
+import org.wso2.carbon.user.api.UserRealm;
+import org.wso2.carbon.user.api.UserStoreException;
+import org.wso2.carbon.user.core.UserStoreManager;
+import org.wso2.carbon.user.core.service.RealmService;
+import org.wso2.carbon.utils.multitenancy.MultitenantUtils;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.util.HashMap;
+
+@PrepareForTest({User.class, IdentityTenantUtil.class, BasicAuthRequestPathAuthenticatorServiceComponent.class, MultitenantUtils.class, AuthenticatedUser.class, FrameworkUtils.class, IdentityUtil.class})
+public class BasicAuthRequestPathAuthenticatorTest  extends PowerMockIdentityBaseTest{
+
+    private BasicAuthRequestPathAuthenticator basicAuthRequestPathAuthenticator;
+    private static final String AUTHENTICATOR_NAME = "BasicAuthRequestPathAuthenticator";
+    private static final String AUTHORIZATION_HEADER_NAME = "Authorization";
+    private String dummyUserName = "testUsername";
+    private String dummyPassword = "testPassword";
+    private int dummyTenantId = -1234;
+    private AuthenticatedUser authenticatedUser;
+
+    @Mock
+    Log mockLog;
+
+    @Mock
+    HttpServletRequest mockRequest;
+
+    @Mock
+    HttpServletResponse mockResponse;
+
+    @Mock
+    AuthenticationContext mockContext;
+
+    BasicAuthRequestPathAuthenticator mockBasicAuthRequestPathAuthenticator = spy(new BasicAuthRequestPathAuthenticator());
```

Comment:
You can use the @Spy annotation here instead

Reply:
On a separate note, isn't this the class you are trying to test? 
Any specific reason as to why you had to spy this?

Reply:
Two doThrow methods use the spy class.